webgpu.h

```diff
@@ -1354,10 +1363,10 @@ typedef void (*WGPUProcShaderModuleSetLabel)(WGPUShaderModule shaderModule, char
 
 // Procs of Surface
 typedef WGPUTextureFormat (*WGPUProcSurfaceGetPreferredFormat)(WGPUSurface surface, WGPUAdapter adapter);
-
-// Procs of SwapChain
-typedef WGPUTextureView (*WGPUProcSwapChainGetCurrentTextureView)(WGPUSwapChain swapChain);
-typedef void (*WGPUProcSwapChainPresent)(WGPUSwapChain swapChain);
+typedef void (*WGPUProcSurfaceConfigure)(WGPUSurface surface, WGPUSurfaceConfiguration const * configuration);
+typedef void (*WGPUProcSurfaceUnconfigure)(WGPUSurface surface);
+typedef WGPUTexture (*WGPUProcSurfaceGetCurrentTexture)(WGPUSurface surface, uint32_t * width, uint32_t * height, WGPUSurfaceStatus * status);
```

Comment:
I'm not sure why we need to return all of these by pointer. `width` and `height` is whatever the user configured and is also available with `wgpuTextureGetWidth/Height`. Status could just be `wgpuSurfaceGetStatus`?

@cwfitzgerald wdyt?

Reply:
for `width` & `height` if I didn't misunderstand, didn't you propose to return them [here](https://github.com/webgpu-native/webgpu-headers/pull/164#discussion_r940317239).

for `status` I think we should return it via pointer to enforce users to explicitly check swapchain status

Reply:
We return this immediately - I would also like to discorage a glGetError kinda interface as much as we can.

Reply:
I assume that by "this" you mean the status? That'd work, although maybe we should support passing `NULL` if you don't care. The pointer is slightly unfortunate but IDK how to make a C API better then this (Maybe the function returns the status and it is the WGPUTexture that's returned by pointer? Not sure).